### PR TITLE
V6 qc additions

### PR DESCRIPTION
**Pull request by chrisAta**
This PR has three major changes:
- Changes to `classify_var_regions.py` so it doesn't fail in cases where two models for the same var_region exist in reads, but one of them is below the threshold of 5,000 reads. Instead, it checks the total number of reads for that region independent of the model, and keeps both models if the total is above 5,000 reads.
- New script: `fastq_suffix_header_check.py`, which checks whether the strand information inside fastq headers match the suffix of file names. This script will be used in the extended QC pipeline
- New script: `library_strategy_check.py`, which checks the output of `assess_mcp_proportions.py` to see if the mean values are above a set threshold of 0.3. If the mean value is above this threshold, then the reads are expected to be AMPLICON, and NOT AMPLICON if below. This threshold was arbitrarily set from checking the values for known Amplicon/WGS/Transcriptomic runs, and making a good guess at a safe threshold. This script will also be used in the extended QC pipeline.

### CODE REVIEW

File: mgnify_pipelines_toolkit/analysis/shared/library_strategy_check.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+
```
Comment (KateSakharova) [inline mgnify_pipelines_toolkit/analysis/shared/library_strategy_check.py:1]: empty line, maybe add shebang here?

File: mgnify_pipelines_toolkit/analysis/shared/library_strategy_check.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    parser.add_argument("-s", "--sample", required=True, type=str, help="Sample ID")
+    parser.add_argument("-o", "--output", required=True, type=str, help="Output")
+
+    args = parser.parse_args()
+
+    INPUT = args.input
+    SAMPLE = args.sample
+    OUTPUT = args.output
+
+    return INPUT, SAMPLE, OUTPUT
```
Comment (KateSakharova) [inline mgnify_pipelines_toolkit/analysis/shared/library_strategy_check.py:22]: the same comments about uppercase

**Review by KateSakharova — CHANGES_REQUESTED** [review]
Thanks Chris!
Only python quibbles from me, sorry

File: mgnify_pipelines_toolkit/analysis/amplicon/classify_var_regions.py [hunk 3]
```diff
@@ -330,17 +330,52 @@ def retrieve_regions(tblout_file_list, outfile_prefix, stats_out, condensed_out,
...
+        models_to_remove = []
+
+        for model, value in multiregion_matches.items():
+            new_value = []
+            for region in value:
+                marker_gene = determine_marker_gene(determine_domain(model))
+                full_region = f'{marker_gene}.{region}'
+                if full_region not in regions_to_remove:
+                    new_value.append(region)
+            if new_value == []:
```
[the remark this answers is not in the document]
Comment (chrisAta) [reply mgnify_pipelines_toolkit/analysis/amplicon/classify_var_regions.py:366]: Didn't know this works! very cool

File: mgnify_pipelines_toolkit/analysis/shared/fastq_suffix_header_check.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+            if counter % 4 == 0: # Only do stuff every four lines to hit the header
+                line = line.decode("ascii").strip()
+                curr_read_strand = line[-2:]
+
+                if curr_read_strand != header_str:
+                    reads_with_err[file].append(line)
+                    reads_with_err["total"].append(1)
+
+    if len(reads_with_err) != 0:
```
[the remark this answers is not in the document]
Comment (chrisAta) [reply mgnify_pipelines_toolkit/analysis/shared/fastq_suffix_header_check.py:80]: Cool that this works but I think this one I will keep as it is, it's easier to understand if it's this explicit IMO

**Review by KateSakharova — APPROVED** [review]

**Pull request merged**